### PR TITLE
feat: allow inviting contact groups

### PR DESCRIPTION
**Pull request by miaulalala**
Fixes https://github.com/nextcloud/calendar/issues/167

To Do

- [x] Duplicate Members - create a single attendee and add the duplicate group ids to the `MEMEBR` property
- [ ] Does it actually send invitation emails
- [x] ~~use `nc:groupname@group` instead of `mailto:` prefix?~~ doesn't work

### CODE REVIEW

**Comment by SebastianKrupinski** [comment]
Hey,

Found an issue, I am selecting a Group with 2 members "Group B"

Screenshot 2024-11-12 200218

But after selecting the "Group B" it only shows one attendee and says there is only one member in the group.

Screenshot 2024-11-12 200247

**Comment by st3iny** [comment]
Is User 1 in Group B and thus de-duplicated?

**Comment by SebastianKrupinski** [comment]
> Is User 1 in Group B and thus de-duplicated? Or maybe the missing user does not have an email address?

Good point! User 1 is not in Group B, but Contact 1 and User 1 do have the same email address. I will test again.

File: src/store/calendarObjectInstance.js [hunk 1]
```diff
@@ -492,14 +491,25 @@ export default defineStore('calendarObjectInstance', {
...
 			if (attendee.attendeeProperty.userType === 'GROUP') {
 				attendee.members.forEach(function(member) {
-					calendarObjectInstance.eventComponent.removeAttendee(member.attendeeProperty)
-					const index = calendarObjectInstance.attendees.indexOf(member)
-					if (index !== -1) {
-						calendarObjectInstance.attendees.splice(index, 1)
+					if (Array.isArray(member.attendeeProperty.member) && member.attendeeProperty.member.length > 1) {
+						const removeIndex = member.attendeeProperty.member.findIndex(function(groupname) {
+							if (groupname === attendee.uri) {
+							  return true
```
Comment (st3iny) [inline src/store/calendarObjectInstance.js:500]:
```suggestion
								return true
```

File: src/components/Editor/Invitees/InviteesListSearch.vue [hunk 2]
```diff
@@ -246,7 +297,7 @@ export default {
 
 				// We do not support GROUPS for now
 				if (principal.calendarUserType === 'GROUP') {
-					return false
+					console.debug(principal)
 				}
```
Comment (st3iny) [inline src/components/Editor/Invitees/InviteesListSearch.vue:301]: Debugging leftover. The whole block can be removed now.

File: src/components/Editor/Invitees/InviteesListSearch.vue [hunk 3]
```diff
@@ -180,6 +196,23 @@ export default {
...
 			})
 		},
+		async getContactGroupMembers(groupName) {
+			let results
+			try {
+				results = await HttpClient.post(linkTo('calendar', 'index.php') + '/v1/autocompletion/groupmembers', {
+					groupName,
+				})
+			} catch (error) {
+				console.debug(error)
```
Comment (st3iny) [inline src/components/Editor/Invitees/InviteesListSearch.vue:206]:
```suggestion
				console.error('Failed to fetch contact group members', error)
```

**Comment by GVodyanov** [comment]
@st3iny Did you test emails being sent out while reviewing? I don't have a mail server setup so I don't know how to do that myself 😅.

Also, maybe you or @SebastianKrupinski could you please do me a favor and fix the merge conflict in `lib/Controller/ContactController.php`, there's quite a lot of stuff going on there and I don't want to mess it up

**Comment by SebastianKrupinski** [comment]
> @st3iny Did you test emails being sent out while reviewing? I don't have a mail server setup so I don't know how to do that myself 😅.
> 
> Also, maybe you or @SebastianKrupinski could you please do me a favor and fix the merge conflict in `lib/Controller/ContactController.php`, there's quite a lot of stuff going on there and I don't want to mess it up

Done

**Review by GVodyanov — APPROVED** [review]
Enough people have had their eyes on this by now I reckon

**Comment by SebastianKrupinski** [comment]
👀

**Pull request merged**